boost/call_stack/detail/gnu/symbol.hpp

```diff
@@ -356,39 +357,66 @@ class library : public boost::call_stack::detail::unique
     {
         pid_t pid = ::getpid();
         std::string procFile = "/proc/" + boost::lexical_cast<std::string>(pid) + "/maps";
-        std::ifstream mapsFile(procFile.c_str());
+        std::ifstream proc_maps_file(procFile.c_str());
         std::string line;
-        
+
         boost::system::error_code ec;
-        std::string filePath =
+        std::string executable_file_path =
             boost::filesystem::canonical(binfile, ec).string();
         if (ec)
         {
-            filePath = std::string(binfile);
-        }            
-        
-        int lineNo = 0;
-        while (std::getline(mapsFile, line))
+            executable_file_path = std::string(binfile);
+        }
+
+        int line_no = 0;
+        while (std::getline(proc_maps_file, line))
         {
-            if (boost::algorithm::ends_with(line, filePath))
+            if (boost::algorithm::ends_with(line, executable_file_path))
             {
-                if (lineNo == 0) // this is the executable itself
+                // this is the executable itself and we don't have a position independent executable
+                if (line_no == 0 && !is_position_independent_executable(executable_file_path))

```

Comment:
This additional check `is_position_independent_executable` and its implementation is the real fix.